src/spikeinterface/core/waveform_extractor.py

```diff
@@ -2016,6 +2032,9 @@ def set_params(self, **params):
         params = self._set_params(**params)
         self._params = params
 
+        if self.waveform_extractor.is_read_only():
```

Comment:
Maybe we should run error instead no ?
set_params on read only should be allowed.

Reply:
It is allowed, but only in mem in this case

Reply:
henestly set_params to a read only should be not aalowed.
What is the use here ?

Reply:
It skips saving to disk!